pkg/operation/botanist/extensions/dns/dnsowner_test.go

```diff
@@ -0,0 +1,137 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dns_test
+
+import (
+	"context"
+	"fmt"
+
+	cr "github.com/gardener/gardener/pkg/chartrenderer"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	. "github.com/gardener/gardener/pkg/operation/botanist/extensions/dns"
+	. "github.com/gardener/gardener/test/gomega"
+
+	dnsv1alpha1 "github.com/gardener/external-dns-management/pkg/apis/dns/v1alpha1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("#DNSOwner", func() {
+	const (
+		deployNS     = "test-chart-namespace"
+		dnsOwnerName = "test-deploy"
+		ownerID      = "owner-id"
+	)
+
+	var (
+		ctrl             *gomock.Controller
+		ca               kubernetes.ChartApplier
+		ctx              context.Context
+		c                client.Client
+		expectedDNSOwner *dnsv1alpha1.DNSOwner
+		vals             *OwnerValues
+		defaultDepWaiter component.DeployWaiter
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+
+		ctx = context.TODO()
+
+		s := runtime.NewScheme()
+		Expect(corev1.AddToScheme(s)).NotTo(HaveOccurred())
+		Expect(dnsv1alpha1.AddToScheme(s)).NotTo(HaveOccurred())
+
+		c = fake.NewFakeClientWithScheme(s)
+
+		vals = &OwnerValues{
+			Name:    "test-deploy",
+			Active:  true,
+			OwnerID: ownerID,
+		}
+
+		expectedDNSOwner = &dnsv1alpha1.DNSOwner{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deployNS + "-" + dnsOwnerName,
+				Namespace: "default",
+			},
+			Spec: dnsv1alpha1.DNSOwnerSpec{
+				OwnerId: ownerID,
+				Active:  pointer.BoolPtr(true),
+			},
+		}
+
+		ca = kubernetes.NewChartApplier(cr.NewWithServerVersion(&version.Info{}), kubernetes.NewApplier(c, meta.NewDefaultRESTMapper([]schema.GroupVersion{})))
+		Expect(ca).NotTo(BeNil(), "should return chart applier")
+
+		defaultDepWaiter = NewDNSOwner(vals, deployNS, ca, chartsRoot(), c)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#Deploy", func() {
+		It("should create correct DNSOwner", func() {
+			Expect(defaultDepWaiter.Deploy(ctx)).ToNot(HaveOccurred())
+
+			actualDNSOwner := &dnsv1alpha1.DNSOwner{}
+			err := c.Get(ctx, client.ObjectKey{Name: deployNS + "-" + dnsOwnerName, Namespace: "default"}, actualDNSOwner)
```

Comment:
`DNSOwner` is currently cluster-scoped.

```suggestion
			err := c.Get(ctx, client.ObjectKey{Name: deployNS + "-" + dnsOwnerName}, actualDNSOwner)
```

Reply:
Yes, I know, but this is the only way I could find to make this test pass, see the explanation above.

Reply:
In fact, I found a way to avoid this by properly configuring the RESTMapper used by the test. Check the latest version of this PR.